### DISCUSSION TITLE
Install app store over .qpkg

### DISCUSSION THREAD

**Discussion by mlndig** [pdulvp/jellyfin-qnap#115] [General]
Hi.

I have had the jellyfin-qnap .qpkg installed manually on my qnap for a while. It's all configured and is running great.
I'd like to use the version from your app-store instead, in order to facilitate updates, but if I try to install it from the repo while the .qpkg has been manually installed and not removed, I get the following message : "[App Center] Failed to install jellyfin. jellyfin already uses this name. Uninstall jellyfin before installing jellyfin.".

I'm a bit hesitant to manually uninstall the .qpkg without fully being sure of what is going to happen. Will I loose all the configuration files and have to restart setting up jellyfin from scratch ?

Thanks in advance.
(Ps : I've seen that questions about the reverse situation have already been asked, but it doesn't explicitly address the question of loosing data and configuration files).

**Comment by NanashiTheNameless** [comment]
Do not, I just tried, lost all the config and user data, backing up and restoring the folder didnt work for me either, for now stay manual... sorry to disappoint

**Reply by NanashiTheNameless** [reply]
TL;DR Don't yet, seems data cannot be easily saved when doing it
hopefully we get a good way to replace it

**Reply by mlndig** [reply]
Thanks for getting back to me so quickly (and sorry for your data)
I will definitely stay manual for now. It's not that big of a hassle.
Hopefully this will prevent more people from wondering the same thing, or worse, to try it and loose their data.

**Reply by NanashiTheNameless** [reply]
eh, none of the videos were lost or anything, just config and metadata as well as logins, but yeah, for now staying manual is the safest

**Reply by pdulvp** [reply]
> Do not, I just tried, lost all the config and user data, backing up and restoring the folder didnt work for me either, for now stay manual... sorry to disappoint

will try to find if there is possible fix for this

**Comment by pdulvp** [comment]
Hi, 
To switch from manual to the one from the store application, the way to do it is: 

With admin account, edit /etc/config/qpkg.conf

in the jellyfin section, add a line
`store = d118acb17dd1c34e962e8c0de3628ba1` (for normal releases (non-prereleases))
save the file.

e.g.
![image](https://github.com/user-attachments/assets/340240c4-6239-4f8e-a760-762c411d4a0c)

Now, on the App Center, press Refresh, and the Installation button shall be replaced by an Update one.

**Reply by mlndig** [reply]
Hi.
Thanks, these steps did work and there is now an "Update" button instead of the old installation button. 
However, when I click the button, stuff seem to happen (it downloads, then updates), but I can't seem to update from 10.10.1 to 10.10.1-a :
It says that I am on 10.10.1, that 10.10.1-a is available and offers to update, but  after I click the update button and it runs its course, apparently with no issue (I have no error message), the version displayed is still 10.10.1 and I'm still offered the option to update, even though the update date was changed to today. Any idea why that might be ? I've tried to reboot my NAS, but it didn't change anything.

**Comment by pdulvp** [comment]
oh yes. :(
i think it's caused by the last release being ending by an "-a" / patch one from the 10.10.1.

same stuff occurred in #99. 
i will dig up as well.

**Reply by mlndig** [reply]
Ah, I see.  I can also just wait until the next "non-letter" release...
Thanks for all your work

**Comment by pdulvp** [comment]
shall be ok if you refresh.

**Reply by mlndig** [reply]
I see that you've renamed the release 10.10.1. 
It doesn't offer the option to update though, I'm guessing because I already have the previous "10.10.1" version installed, so it does not recognize that there is an update with the same name/number
This being said, I just want to mention that I'm perfectly fine waiting for the next release. I'm also ok keeping troubleshooting if that helps.

**Reply by mlndig** [reply]
and now it works, with the new  10.10.1-02 version.
Thanks.

**Comment by pdulvp** [comment]
Hi, 
FYI, In new version, it is now possible to do this manual edition calling `./jellyfin.sh link_to_default_store`
see #117

**Comment by MASTERDR1VE** [comment]
Hi @pdulvp ,

just a small suggestion...
The script `./jellyfin.sh link_to_default_store` is rather "dumb"... it will enter the store id based on the two values in the shell script.
For my system (TS-875A with QuTS hero h5.2.2.2952) the store id was completely different (store = 056edc89f3bea26f35d367224a471b81)

To make it work, a potential solution would be to grab the actual store id from `/etc/config/3rd_pkg_v2.conf`, using the URL as an indicator which store is yours. That's what I did manually.
Otherwise, jellyfin will still be in the default QNAP Store, but within a "3rd party" section, with no option to upgrade.

Regards

**Reply by pdulvp** [reply]
yes as store id on my nas was the same than the one from one log in an issue around here, i supposed it was computed somehow, based on the url. but yes if it defined in a .conf file seems better yes

**Reply by pdulvp** [reply]
done in new release.
